### PR TITLE
[microTVM] Update Zephyr 2.5

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR makes these changes:
- Updates zephyr project on reference virtual machine setup files to 2.5
- Updates zephyr SDK to 12.3
- Adjust required changes to pass zephyr tests on qemu_x86, nucleo and NRF board

Note: tlcpack/zephyr reference machine will have a new release soon based on these changes.

### CODE REVIEW

**Comment by mehrdadh** [comment]
cc @areusch @gromero

File: apps/microtvm/zephyr/demo_runtime/boards/qemu_x86.conf [hunk 1]
```diff
@@ -21,3 +21,4 @@
 CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_TIMER_RANDOM_GENERATOR=y
 
+CONFIG_MAIN_STACK_SIZE=1536
```
Comment (areusch) [inline apps/microtvm/zephyr/demo_runtime/boards/qemu_x86.conf:24]: maybe add a comment here to explain?

**Review by areusch — APPROVED** [review]
thanks @mehrdadh !

**Comment by areusch** [comment]
NOTE: 9941632 passed ci-docker-staging, the pr-merge failure remains identical between 9941632 and 9ffd377, and Vagrantfile changes have no bearing on CI; merging.

**Comment by areusch** [comment]
@mehrdadh ah that's right, can't override the CI even though we're quite sure this won't break. I think the easiest way around this is to try and fix the build error. I think the issue is `k_heap_aligned_alloc` doesn't exist in 2.4.0; turns out, we don't need to use that because the [docs](https://github.com/zephyrproject-rtos/zephyr/blob/5fa12073cb517becba593aecc7229fe02220ef29/include/sys/sys_heap.h#L74) for `sys_heap_alloc` say:

```
The allocated memory is guaranteed to have a starting
 * address which is a multiple of sizeof(void *).  If a bigger alignment
 * is necessary then sys_heap_aligned_alloc() should be used instead.
 *
```

so, want to try changing to `k_heap_alloc` and see if it passes?

**Comment by mehrdadh** [comment]
@areusch changed it.

**Comment by mehrdadh** [comment]
@areusch I think you already approved this. Does it need another reviewer to approve?

**Comment by areusch** [comment]
@mehrdadh nah we should be good. let's merge this.

**Pull request merged**